src/libcore/sync/atomic.rs

```diff
@@ -923,13 +923,14 @@ impl<T> AtomicPtr<T> {
         unsafe { atomic_swap(self.p.get() as *mut usize, ptr as usize, order) as *mut T }
     }
 
-    /// Stores a value into the pointer if the current value is the same as the `current` value.
+    /// Stores a value into the pointer if the current pointer *address* value is the same
+    /// as the `current` pointer *address* value.
     ///
-    /// The return value is always the previous value. If it is equal to `current`, then the value
-    /// was updated.
+    /// The return value is always the previous value contained in the pointer. If it is equal
+    /// to the value pointed by `current`, then the value was updated.
```

Comment:
Here's my attempt.

"Stores the `new` pointer into the `AtomicPtr` if the presently contained pointer address is the same as the address pointed to by the `current` argument.

The return value is always the previous contained pointer. If its value (the address it points to) is the same as `current` then the pointer was updated."